File: python/string/basic.py
# 字符串的基本用法

# 分割
my_string = 'This is a string of words'
print(my_string.split())

# 格式化
name = 'yy'
print('My name is %s' % name)
age = 18
print('Hello %s. You must be at least %i to continue!' % (name, age))
print('Hello %(first_name)s. You must be at least %(age)i to continue!' %
      {"first_name": name, 'age': age})
print('Hello {}. You must be at least {} to continue!'.format(name, age))
print(f'Hello {name}. You are {age} years old')

File: ts/interface/basic.ts
// 接口本本身用于强类型约束，用于编译期间严格约束检查而已
// 一旦编译后的JS代码，运行时没有任何接口信息
// 可以用来进行面向接口的编程
interface Person {
  name: string;
  age: number;
}

// 接口对象属性字段顺序不重要
let p1: Person = { name: "yy", age: 29 };
let p2: Person = {age: 29, name: "tt"};
